waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -128,14 +129,18 @@ validateUserEntity spec =
   case App.auth (snd $ getApp spec) of
     Nothing -> []
     Just auth ->
-      [ GenericValidationError $ "Entity '" ++ userEntityName ++ "' (referenced by app.auth.userEntity) must have an ID field (specified with the '@id' attribute)"
-        | isNothing idFieldType
-      ]
+      case Entity.getIdField userEntity of
+        Nothing -> [userEntityMissingIdFieldError]
+        Just idField ->
+          if Psl.Util.doesPslFieldHaveAttribute "default" idField
+            then []
+            else [userEntityIdFieldMissingDefaultAttrError]
```

Comment:
Nice! Might be tempting to use Either here as a monad? Btw there is even this thing: https://hackage.haskell.org/package/validation .
But probably just not to complicate, it is simple enough for now, I am writing this just to give ideas for the future!